### DISCUSSION TITLE
where to get database tables for the GitHub GraphQL API?

### DISCUSSION THREAD

**Discussion by anlexN** [community/community#28971] [Apps, API and Webhooks]
I have got [Public schema](https://docs.github.com/en/graphql/overview/public-schema). schema is corresponding to database tables. I have heard Github uses MySQL database in the backend, right?
(removed by moderator)

please help me, please tell me. thank you ❤️

**Comment by jsoref** [comment]
Eh. `@`ing a bunch of people is kinda rude.

That GitHub uses a particular database or particular schema for its database is an implementation detail. It isn't part of their public API, unlike /v3 and /v4 (GraphQL). They do periodically reshuffle how their internal system is structured based on either adding new features or experiencing problems that justify such changes.

So, in short, "you don't".

**Comment by tuves** [comment]
Hi @anlexN,

We're glad you're a part of the GitHub Community. And we want to encourage members to ask questions and learn from one another. That said, tagging staff/multiple people will not get quicker responses; quite the opposite, it can end up causing unnecessary noise. Thank you for helping us maintain a productive and tidy community for all our members.

**Comment by MrSarno** [comment]
I’m sorry, but I’m just going to make a point of not answering questions you ask.

Unsolicited spam is not how you get favours from people. Please reconsider the way you treat others on this platform.

**Comment by anlexN** [comment]
1. I don't need GitHub database implementation detail, but I do need GitHub database table schema (or SQL, ER diagram)

**like this**
```sql
CREATE TABLE [IF NOT EXISTS] table_name (
   column1 datatype(length) column_contraint,
   column2 datatype(length) column_contraint,
   column3 datatype(length) column_contraint,
   table_constraints
);
```
**like this**
![image](https://user-images.githubusercontent.com/8686739/184464670-9e10b028-fff9-4956-b077-fff85c4d6022.png)

2. what database is your says particular database based on? MySQL? PostgreSQL? or GitHub-owned database that is not open source?

**Comment by ng-druid** [comment]
I conducted a quick search for **visualizing graph QL schema** and came across [Moon Modeler](https://www.datensen.com/data-modeling/moon-modeler-for-databases.html). It appears with Moon Modeler you can load an existing schema definition and create a full visualization. Might be worth checking out.

There are probably also tools to recreate the schema as a mysql database. At which point you would generate the ER diagram from an application like MySQL Workbench.

**Reply by anlexN** [reply]
I know Moon Modeler.
Maybe I can recreate the database schema, but it is not official.
